### DISCUSSION TITLE
Handling CP2K AIMD Output with Custom Step Frequency for dpdata

### DISCUSSION THREAD

**Discussion by lbyqtttt** [deepmodeling/dpdata#605] [Q&A] [answered]
I've been working with CP2K for AIMD simulations and have run into an issue with processing the output data using dpdata. In my simulations, I've set the output frequency to every 10 steps to reduce the size of the output files. However, dpdata seems to expect data for every step, leading to a mismatch in the expected and actual number of data points.
![image](https://github.com/deepmodeling/dpdata/assets/108402268/e8d18d57-4be5-4e97-9410-a2fd94aed4f9)
Thank you in advance for your assistance.

**Comment by robinzyb** [comment]
cp2kdata is expected to check the consistence for the number of frames for each file, including `-1.ener`, `-pos-1.xyz`, `-frc-1.xyz`, `output`. Setting frequency for output alone is not sufficient. You can modify the `-1.ener` file or other files to be consistent with the `output`

**Reply by lbyqtttt** [reply]
I greatly appreciate your prompt and informative response. Following your advice, I have successfully synchronized the output frequency and handled with dpdata. Thank you once again for your assistance.

**Comment by robinzyb** [comment, marked as answer]
This updated documentation is likely to help you.
https://github.com/robinzyb/cp2kdata/blob/main/docs/dpdata_plugin.md

**Reply by lbyqtttt** [reply]
The problem has been well solved. I am truly grateful for your help.
